### PR TITLE
Add tests for Checkout Terms block

### PR DESCRIPTION
**Pull request by opr**
This PR adds tests for the Checkout Terms inner block, it tests both the editor functionality and the front end block.

I had to install `@wordpress/block-editor` as a dev dependency because this isn't present in a jest environment and we need it to test the behaviour of the blocks in the editor.

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [x] Unit tests
  * [ ] E2E tests
### Manual Testing

How to test the changes in this Pull Request:

1. Ensure tests pass in CI 

### User-facing testing instructions
There are no user-facing testing instructions required for this PR.

### Changelog

> Skip changelog.

### CODE REVIEW

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/edit.js [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+describe( 'Edit', () => {
+	it( 'Renders a checkbox if the checkbox attribute is true', async () => {
+		const { container } = render(
+			<Edit
+				attributes={ {
+					text: 'I agree to the terms and conditions',
+					checkbox: true,
+				} }
+				setAttributes={ () => {} }
```
Comment (alexflorisca) [inline assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/edit.js]: Not a big deal, but [I think we agreed](p1638224334081700/1638213487.071400-slack-C8X6Q7XQU) to use `() => void 0` for empty callback functions as it plays nicely with typescript. In fact the return type should be `void` as [typed here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/edit.tsx#L27)

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		validationData = {
+			hasValidationErrors: false,
+			getValidationError: jest.fn(),
+			clearValidationError: jest.fn(),
+			hideValidationError: jest.fn(),
+			setValidationErrors: jest.fn(),
+		};
+	} );
+
+	it( "Renders a checkbox if the checkbox prop is passed and doesn't if it's not.", async () => {
```
Comment (alexflorisca) [inline assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js]: What do you think about splitting this out into two tests? I found it a bit confusing reading this

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+		const checkbox = await findByLabelText(
+			container,
+			'I agree to the terms and conditions'
+		);
+
+		expect( checkbox ).toBeTruthy();
+		expect( checkbox.getAttribute( 'type' ) ).toEqual( 'checkbox' );
+
+		rerender(
```
Comment (alexflorisca) [inline assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js]: I was expecting something a little different. I was expecting the second render to pass `checkbox={ false }` based on the description of the test. Maybe I misunderstood?

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+			<FrontendBlock
+				checkbox={ true }
+				text={ 'I agree to the terms and conditions' }
+				validation={ validationData }
+			/>
+		);
+		const textOnly = await screen.findByLabelText(
```
Comment (alexflorisca) [inline assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js]: Why are we using `screen.findByLabelText` here and just `findByLabelText` above? Just checking this is intended

**Review by alexflorisca — COMMENTED** [review]
Good work dude, just a few questions and a suggestion from me, nothing major!

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js (hunk 2 above)
Comment (opr) [reply assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js]: Yeah I agree, done!

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js (hunk 4 above)
Comment (opr) [reply assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js]: My bad, removed!

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js (hunk 3 above)
Comment (opr) [reply assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/frontend.js]: Removed this into a different test. This was passing somehow but you're correct it should have been false.

**Comment by opr** [comment]
Thanks for the review @alexflorisca I've made some changes and tidied the code up a bit. Good observations 🙌🏼

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/edit.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+	it( 'Renders a notice if either the terms and conditions or privacy url attribute are unset', async () => {
+		blockSettingsMock.PRIVACY_URL = '';
+		blockSettingsMock.TERMS_URL = '';
+		const { container } = render(
+			<Edit
+				attributes={ {
+					text: 'I agree to the terms and conditions',
+					checkbox: true,
+				} }
+				setAttributes={ () => {} }
```
Comment (alexflorisca) [inline assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/edit.js]: nitpick: `() => void 0` here too. Should have mentioned there are a few occurances of this in my initial review, my bad!

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/edit.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+	it( 'Reminds users to set a URL for their terms and conditions if they are not in the terms textbox', () => {
+		blockSettingsMock.TERMS_URL = '/terms';
+		blockSettingsMock.PRIVACY_URL = '/privacy';
+		const { container } = render(
+			<Edit
+				attributes={ {
+					text: 'I agree to the terms and conditions',
+					checkbox: true,
+				} }
+				setAttributes={ () => {} }
```
Comment (alexflorisca) [inline assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/edit.js]: `() => void 0`

File: assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/edit.js [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		blockSettingsMock.TERMS_URL = '/terms';
+		blockSettingsMock.PRIVACY_URL = '/privacy';
+		const { container } = render(
+			<Edit
+				attributes={ {
+					text:
+						'I agree to the <a href="/terms">terms</a> and <a href="/privacy">privacy</a>',
+					checkbox: true,
+				} }
+				setAttributes={ () => {} }
```
Comment (alexflorisca) [inline assets/js/blocks/cart-checkout/checkout/inner-blocks/checkout-terms-block/test/edit.js]: `() => void 0`

**Review by alexflorisca — APPROVED** [review]
Nice one dude, thanks for making those changes. There's a few more occurrences where it would be good to replace `() => {}` with `() => void 0` but it's a minor nitpick, so good to go from my side 👍

**Pull request merged**